### ISSUE TITLE
Startup Failures are swallowed 

### ISSUE THREAD

**Issue by wjdavis5** [Azure/Azure-Functions#1500]
We have an Azure Function 2.x running on consumption with FrontDoor in ... front.
So it looks like this.

```
                 [Azure Function East 2.x Consumption]     
[FrontDoor] - < 
                 [Azure Function West 2.x Consumption]     

```

We have Azure Monitor configured to send us an Alert anytime Response code 5xx is thrown on the azure function.

 - Since putting Front Door in place, we have been getting 5xx alerts several times a day. 
 - These 5xx are not making into Application Insights. 
 - Nothing in Log analytics about a failure.

Historically we know this can happen when an Azure Function has a Startup failure. However, looking through logs in Kudu, we dont see any startup failures. 

We do see alot of entries where the function run time is starting and stopping, as we would expect to see on a consumption plan function.

SO. We think that these 5xx errors are being caused by Front Door sending a health check to the function when it is starting up. (Front Door is very chatty on health checks)
But we cant prove this b/c the Azure Function Runtime provides 0 insight into startup failures. 

**Question**

- How can I see Azure Function Runtime startup failures?
- What plans, if any, do you have to make this information more easily visible?

In short, lacking the ability to fully investigate service failures is a major deterrent for our continued use of Azure Functions.

**Comment by v-bbalaiagar** [comment]
Hi @wjdavis5, Apologies for the delayed response. Were you able find a solution? do let us know if a follow up is required. I shall follow with the right team members to resolve this issue. The issue was somehow lost in the trace.
